app/models/patient.rb

```diff
@@ -27,6 +27,16 @@ def observations
     @observations
   end
 
+  def weight
+    if @observations.map(&:comment).include?("Body Weight")
```

Comment:
`@observations` gets set only after the `observations` method is called. Instead of referencing `@observations`, try using the `observations` method instead.

I pushed a fix to master just now, fixing the memoization (early return) for #observations and other similar methods.